misk/src/main/kotlin/misk/clustering/DefaultCluster.kt

```diff
@@ -0,0 +1,156 @@
+package misk.clustering
+
+import com.google.common.util.concurrent.AbstractExecutionThreadService
+import com.google.inject.Key
+import misk.DependentService
+import misk.inject.keyOf
+import misk.logging.getLogger
+import org.eclipse.jetty.util.BlockingArrayQueue
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicReference
+import javax.inject.Singleton
+
+/**
+ * A [DefaultCluster] is the default implementation of the [Cluster], which relies on an outside
+ * source such as a cluster watcher to inform it as members become ready or not ready. The
+ * [DefaultCluster] handles computing the actual cluster changes, and managing application watches
+ * that need to be triggered as the cluster changes.
+ */
+@Singleton
+internal class DefaultCluster(
```

Comment:
Dig! Makes me contemplate dropping the `Cluster` interface and just having this

Reply:
... probably not right though ’cause this needs a thread and fakes shouldn’t need a thread!